### PR TITLE
[Ven 1551] [VEN-1552]: Supply fork tests

### PR DESCRIPTION
**Pull request by Debugger022**
## Description

Resolves #
- VEN-1551
- VEN-1552

### CODE REVIEW

File: tests/hardhat/Fork/supply.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+const usdtBorrowAmount = convertToUnit("1", 4);
+
+async function configureTimelock() {
+  impersonatedTimelock = await initMainnetUser(ADMIN, ethers.utils.parseUnits("2"));
+  impersonatedOracleOwner = await initMainnetUser(ORACLE_ADMIN, ethers.utils.parseUnits("2"));
+}
+
+async function configureVToken(vTokenAddress: string) {
+  const VToken = VToken__factory.connect(vTokenAddress, impersonatedTimelock);
+  return VToken;
```
Comment (chechu) [inline tests/hardhat/Fork/supply.ts:66]:
```suggestion
  return VToken__factory.connect(vTokenAddress, impersonatedTimelock);
```

File: tests/hardhat/Fork/supply.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+      // Assert current exchange rate
+      let exchangeRate = await vUSDT.callStatic.exchangeRateCurrent();
+      let calculatedRate = await calculateExchangeRate();
+      expect(exchangeRate).closeTo(calculatedRate, 1);
+
+      // Mint vUSDT with first account(acc1)
+      await usdt.connect(acc1Signer).allocateTo(acc1, convertToUnit(2, 18));
+      await usdt.connect(acc1Signer).approve(vUSDT.address, convertToUnit(2, 18));
+      await expect(vUSDT.connect(acc1Signer).mint(convertToUnit(1, 18))).to.emit(vUSDT, "Mint");
```
Comment (chechu) [inline tests/hardhat/Fork/supply.ts:160]: Any special reason to prepare 2 USDT and then only supply 1 USDT?

File: tests/hardhat/Fork/supply.ts [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+      // Mine 300,000 blocks
+      await mine(blocksToMint);
+
+      // Assert current exchange rate
+      exchangeRate = await vUSDT.callStatic.exchangeRateCurrent();
+      calculatedRate = await calculateExchangeRate();
+      expect(exchangeRate).closeTo(calculatedRate, 1);
+
+      // Set oracle price for usdt
+      await priceOracle.setDirectPrice(usdt.address, convertToUnit("1", 15));
```
Comment (chechu) [inline tests/hardhat/Fork/supply.ts:171]: USDT price is set in the `beforeEach` function. Should it be USDD here?

File: tests/hardhat/Fork/supply.ts [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+      await usdt.connect(acc1Signer).approve(vUSDT.address, convertToUnit(2, 18));
+      await expect(vUSDT.connect(acc1Signer).mint(convertToUnit(1, 18))).to.emit(vUSDT, "Mint");
+
+      // Mine 300,000 blocks
+      await mine(blocksToMint);
+
+      // Assert current exchange rate
+      exchangeRate = await vUSDT.callStatic.exchangeRateCurrent();
+      calculatedRate = await calculateExchangeRate();
+      expect(exchangeRate).closeTo(calculatedRate, 1);
```
Comment (chechu) [inline tests/hardhat/Fork/supply.ts:168]: We could extract this block to a function. It's repeated 6 times

File: tests/hardhat/Fork/supply.ts [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+      await usdt.connect(acc2Signer).approve(vUSDT.address, convertToUnit(2, 18));
+      await expect(vUSDT.connect(acc2Signer).mint(convertToUnit(1, 18))).to.emit(vUSDT, "Mint");
+
+      // Mint vUSDD with second account(acc2)
+      await usdd.connect(acc3Signer).faucet(mintAmount);
+      await usdd.connect(acc3Signer).approve(vUSDD.address, mintAmount);
+      await expect(vUSDD.connect(acc3Signer).mint(mintAmount)).to.emit(vUSDD, "Mint");
+
+      // Borrow usdt with second account(acc2)
+      await expect(vUSDT.connect(acc3Signer).borrow(usdtBorrowAmount)).to.be.emit(vUSDT, "Borrow");
```
Comment (chechu) [inline tests/hardhat/Fork/supply.ts:287]: In the comments we say "second account", but we are really using the acc3

File: tests/hardhat/Fork/supply.ts [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+      // Accural all the interest till latest block
+      await vUSDT.accrueInterest();
+
+      // Assert current exchange rate
+      exchangeRate = await vUSDT.callStatic.exchangeRateCurrent();
+      calculatedRate = await calculateExchangeRate();
+      expect(exchangeRate).closeTo(calculatedRate, 1);
+    });
+
+    it("Two users Mint, one redeems", async () => {
```
Comment (chechu) [inline tests/hardhat/Fork/supply.ts:269]: I would rename to "Three users Mint, one redeems"

File: tests/hardhat/Fork/supply.ts (hunk 1 above)
Comment (defcon022) [reply tests/hardhat/Fork/supply.ts]: Fixed, 4651fb2039bb0f5ad7bc667b2269087b10f7b6f0

File: tests/hardhat/Fork/supply.ts (hunk 2 above)
Comment (defcon022) [reply tests/hardhat/Fork/supply.ts:174]: Yes, funds were used during liquidation.

File: tests/hardhat/Fork/supply.ts (hunk 5 above)
Comment (defcon022) [reply tests/hardhat/Fork/supply.ts]: Fixed, 4651fb2039bb0f5ad7bc667b2269087b10f7b6f0

File: tests/hardhat/Fork/supply.ts (hunk 6 above)
Comment (defcon022) [reply tests/hardhat/Fork/supply.ts]: Fixed, 4651fb2039bb0f5ad7bc667b2269087b10f7b6f0

**Review by GitGuru7 — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Pull request merged**